src/averageAges.js

```diff
@@ -9,17 +9,18 @@
  * To calculate century:
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
- * @param {object[]} people
+ @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filter = people.filter((person) => (century
+    ? Math.ceil(person.died / 100) === century
+    && person.sex === 'm' : person.sex === 'm'));
+  const mapped = filter.map((person) => person.died - person.born);
```

Comment:
1) you can just use reduce here as a method chaining
2) it's better to use more meaningful variable names

Example:
```
const commonMenAge = selectedMen.map(...).reduce()
```